tests/integration/payment-gateway/SV_WC_Payment_Gateway_Payment_Tokens_Handler_Test.php

```diff
@@ -26,6 +26,81 @@ protected function _after() {
 	}
 
 
+	/**
+	 * @see Framework\SV_WC_Payment_Gateway_Payment_Tokens_Handler::delete_token()
+	 */
+	public function test_delete_token() {
+
+		// store a test token
+		$framework_token = new Framework\SV_WC_Payment_Gateway_Payment_Token( '12345', [
+			'type' => 'credit_card',
+			'last_four' => '1111',
+			'exp_month' => '01',
+			'exp_year'  => '20',
+			'card_type' => 'visa',
+		] );
+
+		$this->get_handler()->update_tokens( 1, [ $framework_token->get_id() => $framework_token ] );
+
+		// prepare a mock token with the same ID of the test token
+		$token = \Codeception\Stub::make(
+			Framework\SV_WC_Payment_Gateway_Payment_Token::class,
+			[
+				'get_id' => $framework_token->get_id(),
+				'delete' => \Codeception\Stub\Expected::once(),
+			],
+			$this
+		);
+
+		// test that the token's delete method is called
+		$this->get_handler()->delete_token( 1, $token );
+	}
+
+
+	/**
+	 * @see Framework\SV_WC_Payment_Gateway_Payment_Tokens_Handler::delete_token()
+	 */
+	public function test_delete_token_marks_another_token_as_deafult() {
+
+		$token_id         = '12345';
+		$default_token_id = '45678';
+
+		// store test tokens
+		$tokens = [
+			'12345' => new Framework\SV_WC_Payment_Gateway_Payment_Token( '12345', [
+				'type' => 'credit_card',
+				'last_four' => '1111',
+				'exp_month' => '01',
+				'exp_year'  => '20',
+				'card_type' => 'visa',
+				'default'   => true,
+			] ),
+			'45678' => new Framework\SV_WC_Payment_Gateway_Payment_Token( '45678', [
+				'type' => 'credit_card',
+				'last_four' => '2222',
+				'exp_month' => '01',
+				'exp_year'  => '20',
+				'card_type' => 'visa',
+			] ),
+		];
+
+		$this->get_handler()->update_tokens( 1, $tokens );
+
+		$this->get_handler()->delete_token( 1, $this->get_handler()->get_token( 1, $token_id ) );
+
+		$this->assertNull( $this->get_handler()->get_token( 1, $token_id ) );
+
+		$default_token = $this->get_handler()->get_token( 1, $default_token_id );
+
+		$this->assertInstanceOf( Framework\SV_WC_Payment_Gateway_Payment_Token::class, $default_token );
```

Comment:
can probably skip this assert as well, since it's really testing `get_token()`

Reply:
Removed in d2324fd9eaedfebb5273a93215e21a4b194c0176.